### PR TITLE
Added TH2 proj comparison, TF1 drawing, unity scaling of histos

### PR DESCRIPTION
**Pull request by Marcellocosti**

### CODE REVIEW

File: CompareGraphs.py [hunk 1]
```diff
@@ -39,13 +40,54 @@
...
+            inObj.Rebin(inputCfg['rebin'])
+
         if isinstance(inObj, TH1):
             inObj.SetDirectory(0)
             inObj.Rebin(inputCfg['rebin'])
 
+            if not inObjs:
+                drawTF1low = inObj.GetBinCenter(1)
+                drawTF1upp = inObj.GetBinCenter(inObj.GetNbinsX())
+
```
Comment (danielbattistini) [inline CompareGraphs.py:90]:
```suggestion
```

File: CompareGraphs.py [hunk 2]
```diff
@@ -54,14 +96,44 @@
                 for iBin in range(inObj.GetNbinsX()):
                     inObj.SetBinContent(iBin+1, inObj.GetBinContent(iBin+1) + inputCfg['shift'])
 
+            if('errbarfillstyle' in inputCfg):
+                inObjAsymm = TGraphAsymmErrors(inObj)
+                for iPoint in range(inObj.GetNbinsX()):
+                    errX = inObj.GetBinWidth(iPoint)/4
+                    inObjAsymm.SetPointEXlow(iPoint, errX*2)
+                    inObjAsymm.SetPointEXhigh(iPoint, errX*2)
```
Comment (danielbattistini) [inline CompareGraphs.py:104]:
```suggestion
```

File: CompareGraphs.py [hunk 3]
```diff
@@ -95,9 +167,15 @@
...
     for iObj, (inObj, legend) in enumerate(zip(inObjs, legends)):
         if isinstance(inObj, TGraph):
-            inObj.Draw('same p')
+            if('drawoptsyst' in plot['input'][iObj]):
+                inObj.Draw('same' + plot['input'][iObj]['drawoptsyst'])
+            else:
+                inObj.Draw('same ' + drawOpts[iObj])
+
+        if isinstance(inObj, TF1):
+            inObj.DrawF1(fx1, fx2, "same")
```
Comment (danielbattistini) [inline CompareGraphs.py:176]: fx1 and fx2 are not defined. Please check

File: CompareGraphs.py [hunk 4]
```diff
@@ -95,9 +167,15 @@
 
     for iObj, (inObj, legend) in enumerate(zip(inObjs, legends)):
         if isinstance(inObj, TGraph):
-            inObj.Draw('same p')
+            if('drawoptsyst' in plot['input'][iObj]):
+                inObj.Draw('same' + plot['input'][iObj]['drawoptsyst'])
+            else:
+                inObj.Draw('same ' + drawOpts[iObj])
```
Comment (danielbattistini) [inline CompareGraphs.py:173]: you can put `drawoptsyst` in `drawopt`

File: CompareGraphs.py [hunk 5]
```diff
@@ -109,7 +187,8 @@
                 legend += f';  #mu={inObj.GetMean():.3f}'
             if plot['opt']['leg']['sigma']:
                 legend += f';  #sigma={inObj.GetStdDev():.3f}'
-        leg.AddEntry(inObj, legend, 'lp')
+        if(legend != ''):
```
Comment (danielbattistini) [inline CompareGraphs.py:190]:
```suggestion
        if legend:
```

File: CompareGraphs.py (hunk 3 above)
Comment (Marcellocosti) [reply CompareGraphs.py:173]: They are defined as the xmin and xmax values of the pad when the `DrawFrame` method is called

File: CompareGraphs.py [hunk 6]
```diff
@@ -3,11 +3,12 @@
 import yaml
 from rich import print
 
-from ROOT import TFile, TCanvas, TLegend, TLine, TH1, TGraph, TGraphErrors, TGraphAsymmErrors, TH1D, gStyle, TLatex
+from ROOT import TFile, TCanvas, TLegend, TLine, TH1, TGraph, TGraphErrors, TGraphAsymmErrors, TH1D, gStyle, TLatex, TH2, TF1, TH1F
```
[the remark this answers is not in the document]
Comment (Marcellocosti) [reply CompareGraphs.py]: This is needed for the ChangeUnits method, I moved the import in the script where the function is defined

**Review by danielbattistini — APPROVED** [review]

**Pull request merged**